packages/endpoint/src/endpoint.js

```diff
@@ -13,13 +13,32 @@ function runCompat(endpoint, options) {
   if (endpoint.schema === undefined) endpoint.schema = null;
 }
 
+let CSP = false;
+try {
+  Function();
+} catch (e) {
+  CSP = true;
+}
+
 export default class Endpoint extends Function {
   constructor(fetchFunction, options) {
-    super('return arguments.callee.fetch.apply(arguments.callee, arguments)');
-    if (fetchFunction) this.fetch = fetchFunction;
-    Object.assign(this, options);
+    let self;
+    if (CSP) {
```

Comment:
why go with the complex approach instead of doing the same non-eval thing for everyone?

Reply:
ah, i see you say it won’t work in IE - but neither should extends Function work there :-/

Reply:
It's transpiled. Does es5 inheritance work with function?

Reply:
no. Extending builtins requires native class syntax.

Reply:
oh hmm....so there's really no way to make this work in IE then?

Reply:
not that i know of.